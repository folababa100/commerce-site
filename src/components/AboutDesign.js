import React from 'react';

const AboutDesign = (props) => {
  return (
    <div>
      <p>Adekunbi Adefolarin</p>
    </div>
  )
}

export default AboutDesign;
