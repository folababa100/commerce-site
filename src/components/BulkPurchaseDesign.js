import React from 'react';

const BulkPurchaseDesign = () => {
  return (
    <div>
      Welcome to bulk page
    </div>
  )
}

export default BulkPurchaseDesign;
