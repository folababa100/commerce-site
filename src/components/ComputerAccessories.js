import React from 'react';

const ComputerAccessories = () => {
  return (
    <div>
      <h1>Here is ComputerAccessories</h1>
    </div>
  )
}

export default ComputerAccessories;
