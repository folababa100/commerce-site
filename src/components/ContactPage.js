import React from 'react';
import Header from './Header';
import ContactDesign from './ContactDesign';

const ContactPage = () => {
  return (
    <div>
      <Header />
      <ContactDesign />
    </div>
  )
}

export default ContactPage;
