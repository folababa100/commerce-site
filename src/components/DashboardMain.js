import React from 'react';
import { connect } from "react-redux";
import DashboardItem from './DashboardItem';
import selectProduct from '../selectors/products'

export const DashboardMain = (props) => {
  return (
    <main className="container">
      <div className="container">
        <div className="row">
          {
            props.products.length === 0 ? (
              <p>No products available yet</p>
            ) : (
              props.products.map((product) => {
                return <DashboardItem key={product.id} {...product} />
              })
            )
          }
        </div>
      </div>
    </main>
  )
}

const mapStateToProps = (state) => {
  return {
    products: selectProduct(state.products, state.filters)
  }
}

export default connect(mapStateToProps)(DashboardMain);
