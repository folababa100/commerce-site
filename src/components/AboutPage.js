import React from 'react';
import Header from './Header';
import AboutDesign from './AboutDesign'

const AboutPage = () => {
  return (
    <div>
      <Header/>
      <AboutDesign/>
    </div>
  )
};

export default AboutPage;
