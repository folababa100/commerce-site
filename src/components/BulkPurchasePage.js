import React from 'react';
import BulkPurchaseDesign from './BulkPurchaseDesign';
import Header from './Header';

const BulkPurchasePage = () => {
  return (
    <div>
      <Header/>
      <BulkPurchaseDesign/>
    </div>
  )
}

export default BulkPurchasePage;
