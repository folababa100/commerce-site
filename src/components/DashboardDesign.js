import React from 'react';
import { DashboardMain } from "./DashboardMain";
import DashboardFooter from "./DashboardFooter";

const DashboardDesign = () => {
  return (
    <div>
      <DashboardMain/>
      <DashboardFooter/>
    </div>
  )
}

export default DashboardDesign;
