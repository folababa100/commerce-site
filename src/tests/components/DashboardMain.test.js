import React from 'react';
import { shallow } from "enzyme";
import {DashboardMain} from '../../components/DashboardMain';

test('should render component correctly', () => {
  const wrapper = shallow(<DashboardMain/>);
  expect(wrapper).toMatchSnapshot();
})

