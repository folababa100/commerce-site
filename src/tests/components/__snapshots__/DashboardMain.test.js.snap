// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component correctly with list of products 1`] = `
<main
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-12 mb-4"
    >
      <div
        className="mt-4"
      >
        <div
          className="card"
        >
          <div
            className="card-body"
          >
            <div
              className="row"
            >
              <DashboardItem
                createdAt={0}
                description="A smartphone"
                id={0}
                name="Samsung Phone"
                price={0}
                seller="Slot Limited"
              />
              <DashboardItem
                createdAt={1}
                description="An elegant smartphone"
                id={1}
                name="Iphone Phone"
                price={1}
                seller="Jumia Limited"
              />
              <DashboardItem
                createdAt={2}
                description="A low coat smartphone"
                id={2}
                name="Tecno Phone"
                price={2}
                seller="Unknown Limited"
              />
              <div
                className="fixed-action"
              >
                <button
                  className="btn-floating"
                  style={
                    Object {
                      "outline": "none",
                    }
                  }
                >
                  <i
                    className="ion-edit"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</main>
`;

exports[`should render component with empty products 1`] = `
<main
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-12 mb-4"
    >
      <div
        className="mt-4"
      >
        <div
          className="card"
        >
          <div
            className="card-body"
          >
            <div
              className="row"
            >
              <p>
                No products available yet
              </p>
              <div
                className="fixed-action"
              >
                <button
                  className="btn-floating"
                  style={
                    Object {
                      "outline": "none",
                    }
                  }
                >
                  <i
                    className="ion-edit"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</main>
`;
